server/RemoteHubClient.ts

```diff
@@ -190,4 +202,60 @@ export default class RemoteHubClient extends BaseRemoteClient {
       else callback(null);
     });
   };
+
+  private onImportProject = (data: ImportProjectData) => {
+    // Import a project with a zip archive
+    // 1] Extract the content of the zip file in a temporary folder
+    // 2] Move the extracted content into the projects' folder
+
+    // Extract the content of the zip file in a temporary folder
+    temp.mkdir("project", (err: Error, dirPath: any) => {
+      const inputPath = path.join(dirPath, "project.zip");
+      let rootFolderName: string = null;
+      fs.writeFile(inputPath, data.data, (err) => {
+        if (err) throw err;
+
+        yauzl.open(inputPath, { lazyEntries: true }, (err: Error, zipFile: any) => {
+          if (err != null) throw err;
+
+          zipFile.readEntry();
+          zipFile.on("entry", (entry: any) => {
+            if (rootFolderName === null) {
+              rootFolderName = entry.fileName;
+            }
+
+            if (/\/$/.test(entry.fileName)) {
+              mkdirp(path.join(dirPath, entry.fileName), (err: Error) => {
+                if (err != null) throw err;
+                zipFile.readEntry();
+              });
+            }
+            else {
+              zipFile.openReadStream(entry, (err: Error, readStream: NodeJS.ReadableStream) => {
+                if (err != null) throw err;
+
+                mkdirp(path.dirname(path.join(dirPath, entry.fileName)), (err: Error) => {
+                  if (err != null) throw err;
+
+                  readStream.pipe(fs.createWriteStream(path.join(dirPath, entry.fileName)));
+                  readStream.on("end", () => {
+                    zipFile.readEntry();
+                  });
+                });
+              });
+            }
+          });
+          zipFile.on("end", () => {
+            // move the extracted content into the projects' folder
+            copydir(path.join(dirPath, rootFolderName), path.join(this.server.projectsPath, rootFolderName), (err: Error) => {
+              if (err != null) throw err;
+              this.server.loadProject(rootFolderName, (err: Error, manifest: SupCore.Data.ProjectManifest) => {
+                if (err != null) throw err;
```

Comment:
Here we still need to:

 * Add the project manifest to `this.server.projects` (see here: https://github.com/superpowers/superpowers-core/blob/70f9b5b89d1c504341a725747f15160212aa020b/server/RemoteHubClient.ts#L73). This will make it appear in the list when people reload the page.
 * And then we need to notify all people already connected to the hub of the new project (see https://github.com/superpowers/superpowers-core/blob/70f9b5b89d1c504341a725747f15160212aa020b/server/RemoteHubClient.ts#L122).

Otherwise the new project will be imported but you'll need to restart the server before it can be joined.